solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -1,26 +1,78 @@
+use std::str::FromStr;
+
+use anchor_lang::prelude::{CpiContext, Pubkey, AccountInfo};
 use anchor_lang::solana_program::msg;
+use anchor_lang::solana_program::pubkey::ParsePubkeyError;
+use anchor_spl::token::{spl_token, Burn, MintTo, Transfer};
 use ibc::applications::transfer::context::{
     TokenTransferExecutionContext, TokenTransferValidationContext,
 };
 use ibc::applications::transfer::error::TokenTransferError;
-use ibc::applications::transfer::PrefixedCoin;
+use ibc::applications::transfer::{PrefixedCoin, Amount};
 use ibc::core::ics24_host::identifier::{ChannelId, PortId};
-use ibc::Signer;
+use uint::FromDecStrErr;
+use primitive_types::U256;
 
 // use crate::module_holder::IbcStorage<'_,'_>;
-use crate::IbcStorage;
+use crate::{IbcStorage, MINT_ESCROW_SEED};
+
+pub struct InnerPubkey(pub Pubkey);
+
+impl TryFrom<ibc::Signer> for InnerPubkey {
+
+   type Error = ParsePubkeyError; 
```

Comment:
You can also write this as:

```suggestion
   type Error = <Pubkey as FromStr>::Err;
```